contracts/game/src/lib.cairo

```diff
@@ -306,7 +346,12 @@ mod Game {
 
             // ensure player is not exceeding the rate limit
             let block_number = starknet::get_block_info().unbox().block_number;
-            if !adventurer.block_changed_since_last_action(block_number) {
+
+            // we check if optimistic start and has not started
+            let is_starting = self._starting_entropy.read(adventurer_id) == 0
```

Comment:
Can't see anything that prevents a player from starting, waiting one block, then calling `attack` directly rather than `optimistic_start`.